fact_extractor/plugins/unpacking/srec/code/srec.py

```diff
@@ -0,0 +1,36 @@
+'''
+This plugin decodes / unpacks Motorola SRecord files (.srec)
+'''
+from pathlib import Path
+
+import bincopy
+
+NAME = 'Motorola S-Record'
+MIME_PATTERNS = ['firmware/srecord']
+VERSION = '0.1'
+
+
+def unpack_function(file_path, tmp_dir):
+    '''
+    file_path specifies the input file.
+    tmp_dir should be used to store the extracted files.
+    '''
+    target_file = Path(tmp_dir, Path(file_path).name)
+
+    try:
+        with open(file_path, 'rb') as f:
+            srec = f.readlines()
+        with open(str(target_file.absolute()), 'wb') as f:
+            for line in srec:
+                type_, address, size, data = bincopy.unpack_srec(line.decode())
+                f.write(data)
+    except bincopy.Error as srecError:
```

Comment:
please add another `except` block to catch possible file read errors

Reply:
also, `srecError` ist not a valid variable name (please use lower case with underscores instead of camel case)